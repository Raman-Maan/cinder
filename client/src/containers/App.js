import React, { Component } from 'react';
import logo from '../images/logo.svg';
import './App.css';

class App extends Component {
  state = {
    server_response: ''
  };

  getMessage = () => {
    return fetch('/api/user/example')
      .then(res => res.json())
      .catch(err => console.error(err));
  };

  componentDidMount() {
    this.getMessage()
      .then(msg => this.setState({ server_response: msg.example }))
      .catch(err => console.error(err));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">{this.state.server_response}</p>
      </div>
    );
  }
}

export default App;
