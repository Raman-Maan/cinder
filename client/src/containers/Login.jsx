import React, { Component } from 'react';
import { Container, Form, FormGroup, Label, Input, Button } from 'reactstrap';
import { Redirect, Link } from 'react-router-dom';

import './styles/Login.css';
import logo from '../assets/logo.svg';

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      loggedIn: false,
      loginBtnText: 'Log In'
    };

    this.setInput = this.setInput.bind(this);
    this.userLogin = this.userLogin.bind(this);
  }

  setInput(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  userLogin() {
    // TODO - validate fields

    const payload = { email: this.state.email, password: this.state.password };

    this.setState({ loginBtnText: 'Logging In...' });

    // attempt to log in
    fetch('/api/user/login', {
      method: 'POST',
      body: JSON.stringify(payload)
    })
      .then(res => {
        // all good, let's go to home page
        if (res.status === 200) {
          this.setState({ loginBtnText: 'Log In', loggedIn: true });
        } else {
          throw new Error(`Login Error: ${res.status}`);
        }
      })
      .catch(err => {
        this.setState({ loginBtnText: 'Log In' });
        // TODO - better errors for user - what went wrong?
        switch (err) {
          default:
            console.error(err);
        }
      });
  }

  render() {
    if (this.state.loggedIn) {
      return <Redirect to="/" />;
    }

    return (
      <div className="Login">
        <Container>
          <div className="logo">
            <h1 className="display-3">cinder</h1>
            <img src={logo} className="App-logo" alt="logo" />
          </div>
          <hr />

          <Form className="login-form">
            <FormGroup>
              <Label for="email">Email</Label>
              <Input
                required
                type="email"
                name="email"
                placeholder="Email"
                onChange={this.setInput}
              />
            </FormGroup>

            <FormGroup>
              <Label for="password">Password</Label>
              <Input
                required
                type="password"
                name="password"
                onChange={this.setInput}
                placeholder="Password"
              />
            </FormGroup>

            <FormGroup>
              <Button
                outline
                block
                color="dark"
                type="button"
                onClick={this.userLogin}
              >
                {this.state.loginBtnText}
              </Button>

              <Button outline block color="secondary" tag={Link} to="/signup">
                Sign Up
              </Button>
            </FormGroup>
          </Form>
        </Container>
      </div>
    );
  }
}

export default Login;
