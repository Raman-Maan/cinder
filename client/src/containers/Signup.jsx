import React, { Component } from 'react';
import { Container, Form, FormGroup, Label, Input, Button } from 'reactstrap';
import { Redirect, Link } from 'react-router-dom';

import './styles/Signup.css';
import logo from '../assets/logo.svg';

class Signup extends Component {
  constructor(props) {
    super(props);

    this.setInput = this.setInput.bind(this);
    this.fetchGenderList = this.fetchGenderList.bind(this);
    this.userSignup = this.userSignup.bind(this);

    this.state = {
      genderList: [],
      email: '',
      emailConfirm: '',
      emailCheck: false,
      password: '',
      passwordConfirm: '',
      userName: '',
      birthday: '',
      gender: '',
      signnedUp: false,
      signupBtnText: 'Sign up'
    };
  }

  componentDidMount() {
    this.fetchGenderList();
  }

  fetchGenderList() {
    this.setState({
      genderList: [
        {
          id: 0,
          type: 'Male'
        },
        {
          id: 1,
          type: 'Female'
        },
        {
          id: 2,
          type: 'Other'
        }
      ]
    });
  }

  setInput(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  userSignup(e) {
    e.preventDefault();

    this.setState({ signnedUp: true });
  }

  render() {
    if (this.state.signnedUp) {
      return <Redirect to="/login" />;
    }

    return (
      <div className="Signup">
        <Container>
          <div className="logo">
            <h1 className="display-3">cinder</h1>
            <img src={logo} className="App-logo" alt="logo" />
          </div>
          <hr />

          <div className="inputPart">
            <Form className="signup-form" onSubmit={e => this.userSignup(e)}>
              <div className="row">
                <div className="col-sm">
                  <FormGroup>
                    <Label for="email">Email</Label>
                    <Input
                      required
                      id="email"
                      type="email"
                      name="email"
                      onChange={this.setInput}
                    />
                  </FormGroup>
                </div>
                <div className="col-sm">
                  <FormGroup>
                    <Label for="emailConfirm">Please re-enter email</Label>
                    <Input
                      required
                      id="emailConfirm"
                      type="email"
                      name="emailConfirm"
                      onChange={this.setInput}
                    />
                  </FormGroup>
                </div>
              </div>

              <div className="row">
                <div className="col-sm">
                  <FormGroup>
                    <Label for="password">Password</Label>
                    <Input
                      required
                      id="password"
                      type="password"
                      name="password"
                      onChange={this.setInput}
                    />
                  </FormGroup>
                </div>
                <div className="col-sm">
                  <FormGroup>
                    <Label for="passwordConfirm">
                      Please re-enter password
                    </Label>
                    <Input
                      required
                      id="passwordConfirm"
                      type="password"
                      name="passwordConfirm"
                      onChange={this.setInput}
                    />
                  </FormGroup>
                </div>
              </div>

              <div className="row">
                <div className="col-sm">
                  <FormGroup>
                    <Label for="gender">Gender</Label>
                    <Input
                      type="select"
                      name="gender"
                      id="gender"
                      required
                      onChange={this.setInput}
                    >
                      <option value="" />
                      {this.state.genderList.map(opt => {
                        return (
                          <option key={opt.id} value={opt.id}>
                            {opt.type}
                          </option>
                        );
                      })}
                    </Input>
                  </FormGroup>
                </div>
                <div className="col-sm">
                  <FormGroup>
                    <Label for="birthday">Birthday</Label>
                    <Input
                      required
                      type="date"
                      name="birthday"
                      id="birthday"
                      onChange={this.setInput}
                    />
                  </FormGroup>
                </div>
              </div>

              <FormGroup>
                <Label for="userName">Name</Label>
                <Input
                  required
                  id="userName"
                  name="userName"
                  placeholder="Just first name is enough"
                  onChange={this.setInput}
                />
              </FormGroup>

              <FormGroup>
                <Button id="submitBtn" outline block color="dark" type="submit">
                  {this.state.signupBtnText}
                </Button>

                <Button
                  id="login"
                  outline
                  block
                  color="dark"
                  tag={Link}
                  to="/login"
                >
                  Log in
                </Button>
              </FormGroup>
            </Form>
          </div>
        </Container>
      </div>
    );
  }
}

export default Signup;
