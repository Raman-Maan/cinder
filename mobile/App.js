import React from 'react';
import Expo, { AppLoading } from 'expo';

import Home from './containers/home';

export default class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      fontLoaded: false,
    };
  }

  componentDidMount() {
    this.loadAssets();
  }

  async loadAssets() {
    await Expo.Font.loadAsync({
      Roboto: require('native-base/Fonts/Roboto.ttf'),
      Roboto_medium: require('native-base/Fonts/Roboto_medium.ttf'),
    });

    this.setState({ fontLoaded: true });
  }

  render() {
    if (!this.state.fontLoaded) {
      return <AppLoading />;
    }

    return (
      <Home />
    );
  }
}
