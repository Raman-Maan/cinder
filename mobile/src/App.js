import React from 'react';
import { View, StatusBar } from 'react-native';
import { StackNavigator } from 'react-navigation';

import Home from './containers/home';
import Profile from './containers/blank';
import Matches from './containers/matches';

const RootStack = StackNavigator({
  Home: {
    screen: Home,
  },
  Profile: {
    screen: Profile,
  },
  Matches: {
    screen: Matches,
  },
}, {
  initialRouteName: 'Matches',
  navigationOptions: {
    title: 'cinder',
  },
});

export default class App extends React.Component {
  render() {
    return (
      <RootStack />
    );
  }
}
