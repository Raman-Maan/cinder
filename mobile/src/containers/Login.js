import React from 'react';
import Expo from 'expo';
import { StyleSheet, Text } from 'react-native';
import { Container, H1, Content, Form, Item, Label, Input, Button } from 'native-base';

export default class Login extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
    };
  }

  userLogin() {
    fetch('/api/user/login', {
      method: 'POST',
      body: JSON.stringify({
        email: this.state.email,
        password: this.state.password,
      }),
    }).then((res) => {
      // handle response here
      if (res.status === 200) {
        // all good
        // execute login here
      } else {
        // handle error here
        throw new Error(res.status);
      }
    }).catch(err => err);
  }

  render() {
    return (
      <Container style={styles.container}>
        <Content style={styles.content}>
          <H1 center>cinder</H1>
          <Form>
            <Item floatingLabel>
              <Label>Email</Label>
              <Input onChangeText={txt => this.setState({ email: txt })} />
            </Item>
            <Item floatingLabel>
              <Label>Password</Label>
              <Input secureTextEntry onChangeText={txt => this.setState({ password: txt })} />
            </Item>
            <Button bordered primary block >
              <Text>Log In</Text>
            </Button>
            <Button bordered dark block>
              <Text>Sign Up</Text>
            </Button>
          </Form>
        </Content>
      </Container>
    );
  }
}


const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  content: {
    paddingLeft: 10,
    paddingRight: 10,
  },
});
