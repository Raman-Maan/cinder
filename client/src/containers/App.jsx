import React, { Component } from 'react';
import logo from '../assets/logo.svg';
import './App.css';

class App extends Component {
  static getMessage() {
    return fetch('/api/user/example')
      .then(res => res.json())
      .catch(err => console.error(err));
  }

  constructor() {
    super();
    this.state = {
      server_response: 'No request for message has been made'
    };
  }

  componentDidMount() {
    App.getMessage()
      .then(msg => this.setState({ server_response: msg.example }))
      .catch(err => console.error(err));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">{this.state.server_response}</p>
      </div>
    );
  }
}

export default App;
