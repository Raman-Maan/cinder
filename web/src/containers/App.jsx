import React, { Component } from 'react';
import { Switch, Route, withRouter } from 'react-router-dom';
import PrivateRoute from '../components/PrivateRoute';
import './styles/App.css';
import { connect } from 'react-redux';

import Layout from './Layout';
import Login from './Login';
import Signup from './Signup';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Switch>
          <Route exact path="/login" component={Login} />
          <PrivateRoute
            exact
            path="/"
            authed={this.props.isAuthenticated}
            component={Layout}
          />
          <Route exact path="/signup" component={Signup} />
        </Switch>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
});

export default withRouter(connect(mapStateToProps, null)(App));
