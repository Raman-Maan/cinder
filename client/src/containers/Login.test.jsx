import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import sinon from 'sinon';
import fetchMock from 'fetch-mock';

import App from './App';
import Login from './Login';
import Signup from './Signup';

describe('<Login />', () => {
  it('renders without crashing', () => {
    // simple smoke test
    shallow(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    );
  });

  it('renders correctly', () => {
    const component = renderer
      .create(
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('routes correctly', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/login']}>
        <App />
      </MemoryRouter>
    );
    expect(wrapper.find(Login)).toHaveLength(1);
  });

  describe('signup button', () => {
    const testMatches = [
      {
        userName: 'Some Name 1',
        userID: 1,
        matchDate: '2018-02-03T18:09:05.000Z',
        primaryPic: null
      },
      {
        userName: 'Some Name 2',
        userID: 5,
        matchDate: '2018-01-03T18:09:05.000Z',
        primaryPic: 'http://c.min.ms/m/b/13/13022/e8250eba.jpeg'
      },
      {
        userName: 'Some Name 3',
        userID: 3,
        matchDate: '2018-02-06T18:09:05.000Z',
        primaryPic: null
      },
      {
        userName: 'Some Name 4',
        userID: 2,
        matchDate: '2018-02-04T18:09:05.000Z',
        primaryPic: 'http://c.min.ms/m/b/13/13022/e8250eba.jpeg'
      }
    ];

    beforeEach(() => {
      fetchMock.reset();
      fetchMock.restore();
      fetchMock.get(/\/api\/users\/[0-9]*\/matches/, testMatches);
      fetchMock.get(/\/api\/users\/[0-9]*\/potentials/, []);
    });

    const wrapper = mount(
      <MemoryRouter initialEntries={['/login']}>
        <App />
      </MemoryRouter>
    );
    const signUpBtn = wrapper.find('a#signUpBtn').first();

    it('starts on login page', () => {
      expect(wrapper.find(Login)).toHaveLength(1);
      expect(wrapper.find(Signup)).toHaveLength(0);
    });

    it('redirects to sign up page after click', () => {
      // need {button: 0} when clicking a btn-link
      signUpBtn.simulate('click', { button: 0 });

      expect(wrapper.find(Login)).toHaveLength(0);
      expect(wrapper.find(Signup)).toHaveLength(1);
    });
  });

  describe('submit login', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/login']}>
        <App />
      </MemoryRouter>
    );
    const loginBtn = wrapper.find('button#submitBtn').first();
    const form = wrapper.find('form').first();

    const testFormValues = {
      email: 'foo@email.com',
      password: 'something',
      handleSubmit: jest.fn()
    };

    it('starts on login page', () => {
      expect(wrapper.find(Login)).toHaveLength(1);
      expect(wrapper.find('form.login-form')).toHaveLength(1);
    });

    it('does not submit with empty fields', () => {
      loginBtn.simulate('click');
      expect(wrapper.find(Login)).toHaveLength(1);
    });

    // TODO
    // it('submits with complete fields', () => {
    //   const component = mount(<MemoryRouter><Login {...testFormValues} /></MemoryRouter>);

    //   component.find('input#email').simulate('change', { target: { name: 'email', value: testFormValues.email } });
    //   component.find('input#password').simulate('change', {
    //     target: { name: 'password', value: testFormValues.password },
    //   });
    //   component.find('input#password').simulate('change', {
    //     target: { name: 'password', value: testFormValues.password },
    //   });

    //   component
    //     .find('button#submitBtn')
    //     .first()
    //     .simulate('click');

    //   expect(testFormValues.handleSubmit).toHaveBeenCalledTimes(1);
    // });
  });
});
