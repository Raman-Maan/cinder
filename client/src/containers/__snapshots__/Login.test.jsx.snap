// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<div
  className="Login"
>
  <div
    className="container"
  >
    <div
      className="logo"
    >
      <h1
        className="display-3"
      >
        cinder
      </h1>
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
    </div>
    <hr />
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className=" form-control-label"
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="form-control"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className=" form-control-label"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="form-group"
      >
        <button
          className="btn btn-outline-dark btn-block"
          id="submitBtn"
          onClick={[Function]}
          type="submit"
        >
          Log In
        </button>
        <a
          className="btn btn-outline-secondary btn-block"
          href="/"
          id="signUpBtn"
          onClick={[Function]}
          type={undefined}
        >
          Sign Up
        </a>
      </div>
    </form>
  </div>
</div>
`;
